src/goto-checker/solver_factory.cpp

```diff
@@ -229,9 +232,25 @@ std::unique_ptr<solver_factoryt::solvert> solver_factoryt::get_dimacs()
 
   auto bv_dimacs =
     util_make_unique<bv_dimacst>(ns, *prop, message_handler, filename);
+
   return util_make_unique<solvert>(std::move(bv_dimacs), std::move(prop));
 }
 
+std::unique_ptr<solver_factoryt::solvert> solver_factoryt::get_external_sat()
+{
+  no_beautification();
+  no_incremental_check();
+
+  std::string sat_solver_invocation =
+    options.get_option("sat-solver-invocation");
```

Comment:
I feel it would be nicer if this was an argument, thus pushing the use of `options` 'higher up' in the call stack.  `get_smt2` takes an argument so this would not be entirely inconsistent, also the place where this is called is likely to already reference `options`.

Reply:
`solver_factoryt` relies heavily on `options` so this is fine IMO.